presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -189,12 +185,21 @@ public boolean isLimitGuaranteed(ConnectorSession session)
             return cachedTableHandle;
         }
         Optional<JdbcTableHandle> tableHandle = delegate.getTableHandle(session, schemaTableName);
-        if (tableHandle.isPresent() || cacheMissing) {
+        if ((tableHandle.isPresent() && canBeCached(tableHandle.get())) || (tableHandle.isEmpty() && cacheMissing)) {
             tableHandleCache.put(key, tableHandle);
         }
         return tableHandle;
     }
 
+    private static boolean canBeCached(JdbcTableHandle tableHandle)
```

Comment:
I'm not sure I understand checks within this method. 
@findepi could you please describe within javadocs @return when it returns `true` and when `false`

@kokosing please doublecheck.